src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -3215,6 +3215,55 @@ private static PSConverter<bool> CreateNumericToBoolConverter(Type fromType)
                 valueToConvert.ToString(), resultType.ToString(), exception.Message);
         }
 
+
+        private static Delegate ConvertPSMethodInfoToDelegate(object valueToConvert,
+                                                      Type resultType,
+                                                      bool recurse,
+                                                      PSObject originalValueToConvert,
+                                                      IFormatProvider formatProvider,
+                                                      TypeTable backupTable)
+        {
+            try
+            {
+                var psMethod = (PSMethod)valueToConvert;
+
+                var maybeType = psMethod.instance as Type;
+                var methodInfoCandiates = maybeType != null
+                    ? maybeType.GetMethods(BindingFlags.Static | BindingFlags.Public)
+                    : psMethod.instance.GetType().GetMethods(BindingFlags.Instance | BindingFlags.Public);
+
+                var isAction = resultType.FullName.StartsWith("System.Action`");
+                var comparator = new FuncActionArgsComparator(isAction, resultType.GenericTypeArguments);
+
+                foreach (var candidate in methodInfoCandiates)
+                {
+                    if (candidate.Name != psMethod.Name)
+                    {
+                        continue;
+                    }
+                    if (comparator.SignatureMatches(candidate.ReturnType, candidate.GetParameters()))
+                    {
+                        return maybeType != null
+                            ? candidate.CreateDelegate(resultType)
```

Comment:
I would expect we create the delegate just once - when figuring the conversion - not every time we're asked to convert (e.g. in a loop).